packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -123,9 +180,12 @@ export class EthereumAnchorService implements AnchorService {
 
     if (waitForConfirmation) {
       await lastValueFrom(requestCreated$)
-      return anchorCompleted$.pipe(catchError(errHandler))
+      return anchorCompleted$.pipe(catchError(errHandler), this._updateEvents())
     } else {
-      return concat(requestCreated$, anchorCompleted$).pipe(catchError(errHandler))
+      return concat(requestCreated$, anchorCompleted$).pipe(
+        catchError(errHandler),
+        this._updateEvents()
```

Comment:
given that `_updateEvents()` is also added to the pipeline in `_pollForAnchorResponse()` - does it still need to be added here as well?

Reply:
makes sense, ty